definitions/npm/react-redux_v7.x.x/flow_v0.142.x-/test_useDispatch.js

```diff
@@ -40,4 +42,66 @@ describe('useDispatch', () => {
       );
     }
   });
+
+  it('handles action creator passed in and is typed', () => {
+    function Com() {
+      const dispatch = useDispatch();
+      return (
+        <button
+          onClick={function() {
+            const action = dispatch({ anything: 'string' });
+            (action.anything: string);
+            // $FlowExpectedError[incompatible-cast] will be string because of passed action
+            (action.anything: number);
+          }}
+        >
+          Dispatch time
+        </button>
+      );
+    }
+  });
+
+  it('handles thunks being passed to return the right value', () => {
+    function Com() {
+      const thunkAction = () => (dispatch) => {
+        return '';
+      }
+      const dispatch = useDispatch();
+      return (
+        <div
+          onClick={() => {
+            const a = dispatch(thunkAction());
+            a.toString();
+            // $FlowExpectedError[incompatible-cast] it will return a string, not any which could be cast to number
+            (a: number)
+          }}
+        >
+          Dispatch time
+        </div>
+      );
+    }
+  });
+
+  it('handles thunks defined as a promise to keep their type', (done) => {
```

Comment:
Do we actually need it async for flow to work?... I'm confused.

Reply:
Updated to not be async and just verify it's a promise of given type